clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -233,14 +235,14 @@ LogicalResult CastOp::verify() {
   case cir::CastKind::int_to_bool: {
     if (!resType.isa<mlir::cir::BoolType>())
       return emitOpError() << "requires !cir.bool type for result";
-    if (!(srcType.isInteger(32) || srcType.isInteger(64)))
+    if (!srcType.isa<mlir::IntegerType, mlir::cir::IntType>())
```

Comment:
Since it is still a draft, there is still some work to do in this PR, this included. But I intend to remove the usage of `mlir::IntegerType` wherever possible in this PR.